src/d3a/models/strategy/infinite_bus.py

```diff
@@ -45,33 +44,38 @@ def __init__(self, energy_sell_rate=None, energy_rate_profile=None, energy_buy_r
 
     def _set_market_maker_rate(self):
         if self.energy_rate_profile is not None:
-            GlobalConfig.market_maker_rate = \
-                read_and_convert_identity_profile_to_float(self.energy_rate_profile)
-        elif isinstance(self.energy_rate, (int, float)):
+            GlobalConfig.market_maker_rate = read_arbitrary_profile(
+                InputProfileTypes.IDENTITY, self.energy_rate_profile)
+        elif self.energy_rate is not None:
             GlobalConfig.market_maker_rate = self.energy_rate
-        elif isinstance(self.energy_rate, (str, dict)):
-            GlobalConfig.market_maker_rate = \
-                read_arbitrary_profile(InputProfileTypes.IDENTITY, self.energy_rate)
-        else:
-            GlobalConfig.market_maker_rate = \
-                ConstSettings.GeneralSettings.DEFAULT_MARKET_MAKER_RATE
 
-    def event_activate(self, **kwargs):
+    def _populate_selling_rate(self):
         if self.energy_rate_profile is not None:
-            self.energy_rate = read_and_convert_identity_profile_to_float(self.energy_rate_profile)
+            self.energy_rate = read_arbitrary_profile(InputProfileTypes.IDENTITY,
```

Comment:
I'm wondering why we're dealing with energy_rate as a profile when there is a dedicated variable `energy_rate_profile`, what do you think ?

Reply:
First it was like that even before this PR, second from the external users' perspectve  its more clear to keep both separate but internally why not cast both to a single variable than keeping both?